integration-tests/src/test/scala/tech/beshu/ror/integration/suites/ReindexSuite.scala

```diff
@@ -40,13 +40,25 @@ trait ReindexSuite
   "A reindex request" should {
     "be able to proceed" when {
       "user has permission to source index and dest index"  in {
-        val result = user1ActionManager.actionPost("_reindex", ReindexSuite.reindexPayload("test1_index"))
+        val result = user1ActionManager.actionPost("_reindex", ReindexSuite.reindexPayload("test1_index", "test1_index_reindexed"))
         assertEquals(200, result.getResponseCode)
       }
     }
     "not be able to proceed" when {
-      "user has no permission to source index and dest index"  in {
-        val result = user1ActionManager.actionPost("_reindex", ReindexSuite.reindexPayload("test2_index"))
+      "user has no permission to source index and dest index which are present on ES"  in {
+        val result = user1ActionManager.actionPost("_reindex", ReindexSuite.reindexPayload("test2_index", "test2_index_reindexed"))
+        assertEquals(401, result.getResponseCode)
+      }
+      "user has no permission to source index and dest index which are absent on ES"  in {
+        val result = user1ActionManager.actionPost("_reindex", ReindexSuite.reindexPayload("not_allowed_index", "not_allowed_index_reindexed"))
+        assertEquals(401, result.getResponseCode)
+      }
+      "user has permission to source index and but no permission to dest index"  in {
+        val result = user1ActionManager.actionPost("_reindex", ReindexSuite.reindexPayload("test1_index", "not_allowed_index_reindexed"))
+        assertEquals(401, result.getResponseCode)
+      }
+      "user has permission to dest index and but no permission to source index"  in {
```

Comment:
and one more: user has no access to one of requested source indices

Reply:
Do you mean request with multiple source indices? Is it even possible?

Reply:
no, it isn't. You are right.